### ISSUE TITLE
Cannot create two PeerConnections with outbound video from HoloLens

### ISSUE THREAD

**Issue by kspark-scott** [microsoft/MixedReality-WebRTC#382]
**Describe the bug**
This may be by design, a known issue, or a mistake or misunderstanding on my part, but I'll present it as a bug to get started.

I am experimenting with the new transceiver API to see how we might utilize it.  One goal is to send a single video stream to multiple peers (e.g. two others) by establishing independent PeerConnections with each peer, which I believe is a common WebRTC pattern.  I thought I had seen the scenario discussed in other issues but could not find them.

The problem is I can't find a way to add the local video stream to two different peer connections (details below in reproduction steps).  Is any variation of this scenario possible with the current API, or even with WebRTC in general?

**To Reproduce**
Offering peer is a UWP app running on HoloLens 2.
Both answering peers are UWP apps (the same one) running on Windows 10 PC.

1. Establish video call from HoloLens to PC with SendReceive audio transceiver and SendOnly video transceiver.  This implies successful signalling, call establishment, and playback at answering peer.
2. Attempt to create and configure a second `PeerConnection` for connection to second answer peer.

Variation 1
3. Call `LocalVideoTrack.CreateFromDeviceAsync()` to obtain track to add to new `Transceiver` on new `PeerConnection`.
4. App terminates completely before call returns.

Should this be possible, or are there issues with using the camera in exclusive access mode, or ...?

Variation 2
3. Create new `PeerConnection`, add new video `Transceiver`, and attempt to set `newTransceiver.LocalVideoTrack` to the same `LocalVideoTrack` instance used for the first connection in step 1.
4. Call produces a `System.InvalidOperationException` with the following (very clear) message:
`'Cannot set track (LocalVideoTrack)"experimental video stream" of peer connection (PeerConnection)""
    on transceiver Microsoft.MixedReality.WebRTC.Transceiver of different peer connection (PeerConnection)"".'`

**Expected behavior**
A second `PeerConnection` object can be created with a video transceiver and a local video track that is the same frame source as is being sent to the first answering peer

**SDP messages**
_N/A (signalling is fine, issue is with local track and transceiver configuration)_

**Screenshots**
_N/A_

**Environment**
Offering peer: UWP app on HoloLens 2 @ 10.0.19041.1103
Anwering peer: UWP app on Windows 10 PC @ 10.0.18363.836

It is a single app that runs on both devices.  The app is targeted at 1903 and is built using a MixedReality-WebRTC NuGet package that I built from source from a recent master branch commit (6268f24ca7eab4904eb6576853fe96aaee18746f).

**Comment by djee-ms** [comment]
Hi @kspark-scott,

See #380, I'm in the middle of adding that feature :)

ETA end of week or so I hope, depending on the time it takes to review the various bits. It unfortunately requires some significant breaking changes to extract the sources from the tracks to make them shareable, which changes in particular the class hierarchy on C# and Unity.

**Comment by kspark-scott** [comment]
Thanks @djee-ms , I think I did see that when you created it but didn't understand the significance.  I'm happy to just follow that issue if you would like to close this one, unless you want to use this one to track the success of our use case specifically.

**Comment by djee-ms** [comment]
#380 is a PR; I am happy to leave that issue open as a reference for that work, which was not otherwise logged.

**Comment by djee-ms** [comment]
This is merged to `master`.

**Comment by kspark-scott** [comment]
Hey Jerome (@djee-ms ) and Filippo (@fibann ), commenting on a closed issue just to give you some feedback, for the record.

I just successfully completed the POC I was working on here and am pleased to report that it all worked flawlessly.  I was able to share track sources for both audio and video between two different peer connections.  We are quite excited about the possibilities it opens up for us.  There are of course limitations to the amount of data the HoloLens can process for additional connections, but two video streams at 640x380@30 were quite manageable, about 70% CPU with lots of headroom for 3D rendering.  And the new transceiver API gives us the ability (not yet tested by me, I hope I understand this correctly) to replace a higher resolution track with a lower resolution one on the fly without having to renegotiate the connection.  So we can use the higher quality until we want to add additional participants, at which point we can scale the existing connections back.

So we expect to benefit greatly from all these changes you've been working on recently.  Well done, and thanks!
